soc/arm/microchip_mec/mec1501/dts_fixup.h

```diff
@@ -95,48 +95,9 @@
 #define DT_GPIO_XEC_GPIO240_276_LABEL		DT_MICROCHIP_XEC_GPIO_40081280_LABEL
 #define DT_GPIO_XEC_GPIO240_276_SIZE		DT_MICROCHIP_XEC_GPIO_40081280_SIZE
 
-#ifdef DT_MICROCHIP_XEC_TIMER_40000C00_BASE_ADDRESS
-
-#define DT_COUNTER_MCHP_XEC_0			1
```

Comment:
I don't think it is a good idea to delete the fixups. Remember we struggled with the python code which generates DT_INST_X numbers for GPIO instances. Depending on the python version, the instance numbers varied. The end result is that all the pin configuration and gpios were corrupted.

It is better to keep the fixup

Reply:
The fixups are intended to be removed for things that don't need them.  I get how the GPIO still require such a thing right now because of the importance of know which instance is which.  However for the changes I've made there is no longer anything that needs to keep track of that.

The long term goal has been always to remove the fixups or reduce them drastically.  We'll eventually encode pinmux information in DT which should hopefully reduce the need as well.

For the rare cases that we need to know the specific instance we are going to utilize the nodelabel have an SoC specific nodelabel which would match any vendor docs.  So that would be the plan for GPIO.

Reply:
@galak 
> Remember we struggled with the python code which generates DT_INST_X numbers for GPIO instances. Depending on the python version, the instance numbers varied.

is this still an issue?

Reply:
I think its a combo of which defines got used and how python behaves before version 3.6

Reply:
I'm just leaving this here as a remainder that the python code generating the instances numbers gave different outcomes if ran with python 3.5.1 vs 3.6

https://github.com/zephyrproject-rtos/zephyr/pull/18144
https://github.com/zephyrproject-rtos/zephyr/issues/17751

We cannot have broken FW because of this.

Reply:
This was due to incorrect assumptions about the DT_INST defines.  the python versions just happen to show the issue.  Agreed, we need to be careful here, but ideally in getting drivers to be DT_INST_ based primary and only in rare cases use other defines I think this will be better handled.